src/OnlineSales/Tasks/SyncIPDetailsTask.cs

```diff
@@ -45,12 +51,22 @@ internal override void ExecuteLogTask(List<ChangeLog> nextBatch)
 
             if (!string.IsNullOrEmpty(newIp))
```

Comment:
Let's think about what is going to happen here:

1) Imagine nextBatch.Count == 50, so you have 50 records in your batch
2) On every records you will execute GetIpIfNotExist which as the result will make a separate select call to the database to understand if IP address exists or not.
3) As the result in case of nextBatch.Count == 50 you will make **50** separate calls to the database which is very bad, imagine if batch size is 100 or 1000.... You will generate crazy amount of call to database just to check if IP is there or not.  
4) Then you call dbContext.SaveChanges() inside the loop which means you make separate save calls to the database for every new IP. This will also generate lots of separate calls which is again - very bad. It is not only bad because of the big amount of calls, it is also bad because the process may fail in-between, then some records could be already saved and some were not yet processed. It would not be such in issues in this particular case, but in other scenarios it could be a big problem. Simple moving dbContext.SaveChanges() outside the loop will convert it to batch all or nothing save call which is good.